pkg/shell/vm.go

```diff
@@ -45,16 +47,6 @@ func Run(command string) {
 // maps, and if found, executes the command function and reports any
 // returned error. Otherwise, it runs the command as an external command
 // and returns any raised error.
-func dispatch(command string, args []string) error {
-	function, exists := special.Commands[command]
-	if exists {
-		return function(args)
-	}
-
-	err := runners.Builtin(command, args)
-	if _, is := err.(*runners.NotBuiltinCmd); !is {
-		return err
-	}
-
-	return runners.External(command, args)
+func dispatch(cmd command.Command) error {
```

Comment:
Also not needed in this new design but kept it because I didn't want to break things 😂 will remove it once lexer is done.